packages/app/components/stack-modal/StackModal.tsx

```diff
@@ -74,17 +76,36 @@ export const StackModal = ({
   const lastSlot = orderSlots[orderSlots.length - 1];
   const nextSlot = orderSlots[totalOrderSlotsDone(stackOrder)];
 
-  const remainingFunds =
-    convertedAmount(stackOrder.amount, stackOrder.sellToken.decimals) -
-    totalFundsUsed(stackOrder);
+  const getConfirmCancelContent = () => {
+    if (stackIsFinishedWithFunds(stackOrder))
+      return confirmCancelContent.finishedWithFunds;
 
-  const remainingFundsDescription =
-    remainingFunds > 0 &&
-    `remaining funds, ${remainingFunds} ${stackOrder.sellToken.symbol}`;
+    return confirmCancelContent.notComplete;
+  };
+
+  const remainingFundsText = () =>
+    `The ${stackRemainingFundsWithTokenText(
+      stackOrder
+    )} will be sent to your wallet.`;
 
-  const stackIsComplete =
-    totalOrderSlotsDone(stackOrder) === orderSlots.length &&
-    remainingFunds === 0;
+  const confirmCancelContent = {
+    notComplete: {
+      title: "Are you sure you want to cancel stacking?",
+      description: remainingFundsText(),
+      button: {
+        action: "secondary",
+        text: "Cancel Stacking",
+      },
+    },
+    finishedWithFunds: {
+      title: "Proceed with cancelation to withdraw funds",
+      description: remainingFundsText(),
+      button: {
+        action: "primary",
+        text: `Withdraw ${stackRemainingFundsWithTokenText(stackOrder)}`,
+      },
+    },
+  };
```

Comment:
This can be 2 separate variables. I see no reason to have 1 variable with 2 distinct keys

Reply:
It can

Reply:
fixed.